csharp/src/Ice/IObjectPrx.cs

```diff
@@ -494,4 +525,308 @@ protected ObjectPrx(SerializationInfo info, StreamingContext context)
             IceReference = communicator.CreateReference(str, null);
         }
     }
+
+    // publicly visible Ice-internal class used as base class for request
+    // with non void return value
+    public class OutgoingRequestWithReturnValue<TOutParams>
```

Comment:
Since this is a helper class use by the derived classes below, I think the ctor and various methods should be `private protected` not `internal` or `public`.

Reply:
This need to be public, can not be less visible than the derived types.

Reply:
But all the methods could be `private protected`?

Reply:
yes methods can be `private protected` fixed